sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Expression.scala

```diff
@@ -323,6 +323,20 @@ trait RuntimeReplaceable extends UnaryExpression with Unevaluable {
   // two `RuntimeReplaceable` are considered to be semantically equal if their "child" expressions
   // are semantically equal.
   override lazy val canonicalized: Expression = child.canonicalized
+
+  override def innerChildren: Seq[Expression] = sys.error("RuntimeReplaceable must implement" +
```

Comment:
Since compilers cannot catch the case where one forget to override this, how about it like this?
```
  // RuntimeReplaceable
  def exprsReplaced: Seq[Expression]
  override def innerChildren: Seq[Expression] = exprsReplaced

  // e.g., ParseToTimestamp
  override def exprsReplaced: Seq[Expression] = Seq(Some(left), format).flatten

```

Reply:
Good Point!